sceptre/diffing/diff_writer.py

```diff
@@ -0,0 +1,177 @@
+import json
+from abc import abstractmethod
+from datetime import datetime
+from typing import TextIO, Generic, List
+
+import cfn_flip
+import yaml
+from deepdiff import DeepDiff
+
+from sceptre.diffing.stack_differ import StackConfiguration, StackDiff, DiffType
+
+deepdiff_json_defaults = {
+    datetime.date: lambda x: x.isoformat(),
+    StackConfiguration: lambda x: x._asdict()
+}
+
+
+class DiffWriter(Generic[DiffType]):
+    STAR_BAR = '*' * 80
+    LINE_BAR = '-' * 80
+
+    def __init__(self, stack_diff: StackDiff, output_stream: TextIO, output_format: str):
+        self.stack_name = stack_diff.stack_name
+        self.stack_diff = stack_diff
+
+        self.template_diff = stack_diff.template_diff
+        self.config_diff = stack_diff.config_diff
+        self.is_deployed = stack_diff.is_deployed
+
+        self.output_stream = output_stream
+        self.output_format = output_format
+
+    @property
+    def has_difference(self) -> bool:
+        return self.has_config_difference or self.has_template_difference
+
+    def write(self):
+        self._output(self.STAR_BAR)
+        if not self.has_difference:
+            self._output(f"No difference to deployed stack {self.stack_name}")
+            return
+
+        self._output(f"--> Difference detected for stack {self.stack_name}!")
+
+        if not self.is_deployed:
+            self._write_new_stack_details()
+            return
+
+        self._output(self.LINE_BAR)
+        self._write_config_difference()
+        self._output(self.LINE_BAR)
+        self._write_template_difference()
+
+    def _write_new_stack_details(self):
+        stack_config_text = self._dump_stack_config(self.stack_diff.generated_config)
+        self._output(
+            'This stack is not deployed yet!',
+            self.LINE_BAR,
+            'New Config:',
+            '',
+            stack_config_text,
+            self.LINE_BAR,
+            'New Template:',
+            '',
+            self.stack_diff.generated_template
+        )
+        return
+
+    def _output(self, *lines: str):
+        lines_with_breaks = [f'{line}\n' for line in lines]
+        self.output_stream.writelines(lines_with_breaks)
+
+    def _dump_stack_config(self, stack_config: StackConfiguration) -> str:
+        """Outputs a StackConfiguration to json or yaml, depending on the configured output_format
+
+        Args:
+            stack_config: The StackConfiguration to serialize
+
+        Returns:
+            The jsonified stack config, if json is the output format; otherwise yaml
```

Comment:
You are using a different style of docstring from the rest of the project here. Not sure if people care about such details.

Reply:
Oh yeah, some of this got copied fr my code base which uses google-style docstrings. I'll need to go through this whole pr and do proper restructured style on all the things.